WooCommerce/Classes/ViewRelated/Reviews/ReviewsViewController.swift

```diff
@@ -478,16 +480,14 @@ private extension ReviewsViewController {
 //
 private extension ReviewsViewController {
 
-    /// Enables/disables the navbar buttons if needed
-    ///
-    /// - Parameter filterEnabled: If true, the filter navbar buttons is enabled; if false, it's disabled
-    ///
     func updateNavBarButtonsState() {
-        updateMarkAllReadButtonState()
+        updateMenuButtonState()
     }
 
-    func updateMarkAllReadButtonState() {
-        rightBarButton.isEnabled = viewModel.hasUnreadNotifications
+    /// Show the rightBarButtonItem only if there are unread reviews available.
+    ///
+    func updateMenuButtonState() {
+        navigationItem.rightBarButtonItem = viewModel.hasUnreadNotifications ? rightBarButton : nil
     }
```

Comment:
This method can actually be merged with `updateNavBarButtonsState()` for simplicity:
```suggestion
    func updateNavBarButtonsState() {
        navigationItem.rightBarButtonItem = viewModel.hasUnreadNotifications ? rightBarButton : nil
    }
```
WDYT?

Reply:
I assume that the original idea behind this is to have the ability to introduce separate methods for each navigation bar button item. 

I am OK with merging this into a single method if you want to do it. Kindly let me know.

Reply:
I'm not sure yet if we are going to have new button items there so it's best to keep things simple for now. Since we intend to update only the right bar button item here, how about merging the two methods into one and naming it `updateRightBarButtonItem()` to be more specific?

Reply:
Tagging @selanthiraiyan just in case you miss this thread 😄

Reply:
I agree. Done in https://github.com/woocommerce/woocommerce-ios/pull/5833/commits/859e44fcbb80bc2ca5668976f39df4b791c215f5